master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -180,7 +187,9 @@ public void updateWorkerHeartbeatMeta(
         && (((disks.isEmpty() || healthyDiskNum <= 0) && !conf.hasHDFSStorage()) || highWorkload)) {
       LOG.debug("Worker: {} num total slots is 0, add to excluded list", worker);
       excludedWorkers.add(worker);
-    } else if ((availableSlots.get() > 0 || conf.hasHDFSStorage()) && !highWorkload) {
+    } else if ((availableSlots.get() > 0 || conf.hasHDFSStorage())
+        && !highWorkload
+        && !manuallyExcludedWorkers.contains(worker)) {
```

Comment:
It seems no need to check manuallyExcludedWorkers now.

Reply:
@RexXiong, I have removed this unnecessary check.